src/main/java/net/guides/springboot2/springboot2jpacrudexample/controller/EmployeeController.java

```diff
@@ -2,65 +2,51 @@
 
 import net.guides.springboot2.springboot2jpacrudexample.exception.ResourceNotFoundException;
 import net.guides.springboot2.springboot2jpacrudexample.model.Employee;
-import net.guides.springboot2.springboot2jpacrudexample.repository.EmployeeRepository;
+import net.guides.springboot2.springboot2jpacrudexample.service.EmployeeService;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
 import javax.validation.Valid;
-import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-import java.util.Optional;
 
 @RestController
 @RequestMapping("/api/v1")
 public class EmployeeController {
     @Autowired
-    private EmployeeRepository employeeRepository;
+    private EmployeeService employeeService;
 
     @GetMapping("/employees")
     public List<Employee> getAllEmployees() {
-        List<Employee> employees = employeeRepository.findAll();
+        List<Employee> employees = employeeService.getAllEmployees();
         return employees;
     }
 
     @GetMapping("/employees/{id}")
     public ResponseEntity<Employee> getEmployeeById(@PathVariable(value = "id") Long employeeId) throws ResourceNotFoundException {
-        Optional<Employee> employeeOptional = employeeRepository.findById(employeeId);
-        Employee employee = employeeOptional.orElseThrow(() -> new ResourceNotFoundException("Employee not found for this id :" + employeeId));
-        return ResponseEntity.of(employeeOptional);
+        ResponseEntity<Employee> employeeOptional = employeeService.getEmployeeById(employeeId);
+        return employeeOptional;
     }
 
     @PostMapping("/employees")
     public Employee createEmployee(@Valid @RequestBody Employee employee) {
-        employee = employeeRepository.save(employee);
+        employee = employeeService.createEmployee(employee);
         return employee;
     }
 
     @PutMapping("/employees/{id}")
     public ResponseEntity<Employee> updateEmployee(@PathVariable(value = "id") Long employeeId,
                                                    @Valid @RequestBody Employee employeeDetails) throws ResourceNotFoundException {
-        Employee employee = employeeRepository.findById(employeeId)
-                .orElseThrow(() -> new ResourceNotFoundException("Employee not found for this id : " + employeeId));
-        employee.setEmailId(employeeDetails.getEmailId());
-        employee.setLastName(employeeDetails.getLastName());
-        employee.setFirstName(employeeDetails.getFirstName());
-
-        final Employee updatedEmployee = employeeRepository.save(employee);
-        return ResponseEntity.ok(updatedEmployee);
+        ResponseEntity<Employee> employee = employeeService.updateEmployee(employeeId, employeeDetails);
```

Comment:
Rename employee variable as it is of type ResponseEntity. You cane use names like employeeResponse or something

Reply:
done